src/index.js

```diff
@@ -320,70 +467,74 @@ const initialize = async () => {
       contractStatus.innerHTML = 'Deploying';
 
-      try {
-        contract = await piggybankFactory.deploy();
-        await contract.deployTransaction.wait();
-      } catch (error) {
-        contractStatus.innerHTML = 'Deployment Failed';
-        throw error;
-      }
-
-      if (contract.address === undefined) {
-        return;
-      }
+      await deployPiggyBankContract();
 
-      console.log(
-        `Contract mined! address: ${contract.address} transactionHash: ${contract.transactionHash}`,
-      );
       contractStatus.innerHTML = 'Deployed';
       depositButton.disabled = false;
       withdrawButton.disabled = false;
 
-      depositButton.onclick = async () => {
-        contractStatus.innerHTML = 'Deposit initiated';
-        const result = await contract.deposit({
-          from: accounts[0],
-          value: '0x3782dace9d900000',
-        });
-        console.log(result);
-        contractStatus.innerHTML = 'Deposit completed';
-      };
+      // Deposit button functionality with different library
+      if (libraryInUse === 'ethers') {
+        depositButton.onclick = async () => {
+          contractStatus.innerHTML = 'Deposit initiated';
+          const result = await piggyBankContract.deposit({
+            from: accounts[0],
+            value: '0x3782dace9d900000',
+          });
+          console.log(result);
+          contractStatus.innerHTML = 'Deposit completed';
+        };
+      } else {
+        depositButton.onclick = async () => {
+          contractStatus.innerHTML = 'Deposit initiated';
+          await piggyBankContract.methods
+            .deposit()
+            .send({
+              from: accounts[0],
+              value: '0x3782dace9d900000',
+            })
+            .then((receipt) => {
+              console.log(receipt);
+            });
+          contractStatus.innerHTML = 'Deposit completed';
+        };
+      }
 
-      withdrawButton.onclick = async () => {
-        const result = await contract.withdraw('0xde0b6b3a7640000', {
-          from: accounts[0],
-        });
-        console.log(result);
-        contractStatus.innerHTML = 'Withdrawn';
-      };
+      // Withdraw button functionality with different library
+      if (libraryInUse === 'ethers') {
+        withdrawButton.onclick = async () => {
+          const result = await piggyBankContract.withdraw('0xde0b6b3a7640000', {
+            from: accounts[0],
+          });
+          console.log(result);
+          contractStatus.innerHTML = 'Withdrawn';
+        };
+      } else {
+        withdrawButton.onclick = async () => {
+          await piggyBankContract.methods
+            .withdraw('0xde0b6b3a7640000')
+            .send({
+              from: accounts[0],
+            })
+            .then((receipt) => {
+              console.log(receipt);
+            });
+          contractStatus.innerHTML = 'Withdrawn';
+        };
+      }
 
-      console.log(contract);
+      console.log(piggyBankContract);
```

Comment:
This can be removed.

Reply:
Fixed in: 91555c04c8088ee472c87cc25843642994ec92b4